src/EditorFeatures/Test/Snippets/RoslynLSPSnippetConvertTests.cs

```diff
@@ -0,0 +1,260 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Snippets;
+using Microsoft.CodeAnalysis.Test.Utilities;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Test.Utilities;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.Editor.UnitTests.Snippets
+{
+    [UseExportProvider]
+    public class RoslynLSPSnippetConvertTests
+    {
+        #region Edgecase extend TextChange tests
+
+        [Fact, Trait(Traits.Feature, Traits.Features.RoslynLSPSnippetConverter)]
+        public Task TestExtendSnippetTextChangeForwardsForCaret()
+        {
+            var markup =
+@"[|if ({|placeholder:true|})
+{
+}|] $$";
+
+            var expectedLSPSnippet =
+@"if (${1:true})
+{
+} $0";
+            MarkupTestFile.GetPositionAndSpans(markup, out var outString, out var cursorPosition, out IDictionary<string, ImmutableArray<TextSpan>> dictionary);
+            var stringSpan = dictionary[""].First();
+            var textChange = new TextChange(new TextSpan(stringSpan.Start, 0), outString[..stringSpan.Length]);
+            var placeholders = dictionary["placeholder"].Select(span => span.Start).ToImmutableArray();
+            return TestAsync(markup, expectedLSPSnippet, cursorPosition, ImmutableArray.Create(new SnippetPlaceholder("true", placeholders)), textChange);
```

Comment:
this code looks identical across all your tests.  this *is* your test method that all these tests should actually call into.